api/nodes/views.py

```diff
@@ -106,8 +118,132 @@ def get_queryset(self):
+                valid_methods['folder'].append('DELETE')
+
+        return valid_methods
+
+    @staticmethod
+    def get_file_item(item, valid_file_methods, node_id, cookie, obj_args):
+        file_item = {
+            'valid_self_link_methods': valid_file_methods[item['kind']],
+            'provider': item['provider'],
+            'path': item['path'],
+            'name': item['name'],
+            'node_id': node_id,
+            'cookie': cookie,
+            'args': obj_args,
+            'waterbutler_type': 'file',
+            'item_type': item['kind'],
+        }
+        if file_item['item_type'] == 'folder':
+            file_item['metadata'] = {}
+        else:
+            file_item['metadata'] = {
+                'content_type': item['contentType'],
+                'modified': item['modified'],
+                'size': item['size'],
+                'extra': item['extra'],
+            }
+        return file_item
+
+    def get_queryset(self):
+        query_params = self.request.query_params
+
+        addons = self.get_node().get_addons()
+        user = self.request.user
+        cookie = user.get_or_create_cookie() if self.request.user else None
+        node_id = self.get_node()._id
+        obj_args = self.request.parser_context['args']
+
+        provider = query_params['provider'] if 'provider' in query_params else None
+        path = query_params['path'] if 'path' in query_params else '/'
+        files = []
+
+        if provider is None:
+            valid_self_link_methods = self.get_valid_self_link_methods(user, True)
+            for addon in addons:
+                if addon.config.has_hgrid_files:
+                    files.append({
+                        'valid_self_link_methods': valid_self_link_methods,
+                        'provider': addon.config.short_name,
+                        'name': addon.config.short_name,
+                        'path': path,
+                        'node_id': node_id,
+                        'cookie': cookie,
+                        'args': obj_args,
+                        'waterbutler_type': 'data',
+                        'item_type': 'folder',
+                        'metadata': {},
+                    })
+        else:
+            valid_self_link_methods = self.get_valid_self_link_methods(user, False)
+            url = waterbutler_url_for('data', provider, path, self.kwargs['pk'], node_id, obj_args)
+            waterbutler_request = requests.get(url)
+            waterbutler_data = waterbutler_request.json()['data']
+            if isinstance(waterbutler_data, list):
+                for item in waterbutler_data:
+                    file = self.get_file_item(item, valid_self_link_methods, node_id, cookie, obj_args)
+                    files.append(file)
+            else:
+                files.append(self.get_file_item(waterbutler_data, valid_self_link_methods, node_id, cookie, obj_args))
+
+        return files
+
+    def get_current_user(self):
```

Comment:
This is unnecessary and incorrect. The current user is always accessible through `self.request.user`. Also, view objects don't have a `context` attribute (Serializers do).